### PR TITLE
'as of' date-time formatting

### PR DESCRIPTION
**Pull request by shapesandobjects**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,10 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
+    "moment": "^2.29.1",
```
Comment (chrisahardie) [inline package.json:11]: There are better alternatives to moment, even the authors suggest looking for something else: https://momentjs.com/docs/#/-project-status/

**Review by johnhok — COMMENTED** [review]
Screen Shot 2021-05-03 at 4 16 37 PM

A couple of suggestions.
1. Think we should display on the 12 hour clock for readability.
2. We should probably display a timezone if this is going to be viewed by people across the region so it's clear what timezone we're showing the times in?

File: package.json (hunk 1 above)
Comment (chrisahardie) [reply package.json:11]: Yeah, it's def lightweight - moment is actually a MASSIVE dependency

**Comment by chrisahardie** [comment]
> Screen Shot 2021-05-03 at 4 16 37 PM
> 
> A couple of suggestions.
> 
> 1. Think we should display on the 12 hour clock for readability.
> 2. We should probably display a timezone if this is going to be viewed by people across the region so it's clear what timezone we're showing the times in?

Yeah, the 24 hour clock and `PM` don't work together. 

Sigh... I was wondering how long it would be before we had to get into a conversation about timezones :)

**Comment by picuron** [comment]
> > Screen Shot 2021-05-03 at 4 16 37 PM
> > A couple of suggestions.
> > 
> > 1. Think we should display on the 12 hour clock for readability.
> > 2. We should probably display a timezone if this is going to be viewed by people across the region so it's clear what timezone we're showing the times in?
> 
> Yeah, the 24 hour clock and `PM` don't work together.
> 
> Sigh... I was wondering how long it would be before we had to get into a conversation about timezones :)

**Comment by picuron** [comment]
Wrong button, sorry ^

**Comment by picuron** [comment]
And yeah I didnt even think of timezones. Does `date-fns` do any timezone conversions based on location, or would something else have to be used for that?

**Comment by chrisahardie** [comment]
> And yeah I didnt even think of timezones. Does `date-fns` do any timezone conversions based on location, or would something else have to be used for that?

Yeah. there's an extension for that. We would read in the browser's timezone and then make the calculation to display the correct time with the timezone name - I am assuming the backend is storing in UTC? I'll confirm.

**Comment by shapesandobjects** [comment]
Re-working this using date-fns-tz, so we'll have localized time zone formatting, etc.
Only hold-up atm is that our backend is returning naive datetime objects, which can be local time, not utc specific.  I don't know if this means client or server local time, so will get a PR into the backend to send as utc, and we can be confident of appropriate format handling on the frontend

**Comment by chrisahardie** [comment]
> Re-working this using date-fns-tz, so we'll have localized time zone formatting, etc.
> Only hold-up atm is that our backend is returning naive datetime objects, which can be local time, not utc specific. I don't know if this means client or server local time, so will get a PR into the backend to send as utc, and we can be confident of appropriate format handling on the frontend

Awesome, that's the ticket - remove any and all ambiguity by getting a date with TZ or offset. 👍 👍 👍

**Comment by shapesandobjects** [comment]
Updated to use `date-fns-tz`.  Force pushed to overwrite it all from scratch.

From testing with mock data, I can confirm that this adjusts a UTC time to the user's local system time.  Until the backend gets updated, there might be wonky times shown because of the half-fineshed end-to-end fix, but this half should be done.

**Review by chrisahardie — COMMENTED** [review]
Could I ask that you improve a couple of my tests while you're touching this? We can now assert we are properly displaying dates in `PharmacyList.test.tsx`

Just ensure we can find the datetime in the proper format in the first couple tests:

```
...
await screen.findByRole("heading", { name: /stittsville ida/i });
await screen.findByText(/appointments available/i);
await screen.findByText(/as of may 1 2021, 5:07 pm edt/i);
...
```

```
await screen.findByRole("heading", { name: /stittsville ida/i });
await screen.findByText(/appointments not available/i);
await screen.findByText(/as of may 1 2021, 5:07 pm edt/i);
```

Is it weird to anyone else that the month isn't capitalized, and the timezone isn't uppercase?

**Review by johnhok — APPROVED** [review]
Looks great! 🔥

**Comment by chrisahardie** [comment]
@shapesandobjects Sorry, seems like I made your life harder - tests failed, the markup being rendered is actually `May 1 2021, 9:07 PM UTC`

**Comment by shapesandobjects** [comment]
> @shapesandobjects Sorry, seems like I made your life harder - tests failed, the markup being rendered on the CI server is actually `May 1 2021, 9:07 PM UTC`
> 
> I think this should do it:
> 
> ```
> await screen.findByText(/as of may 1 2021, 9:07 pm utc/i);
> ```
> 
> You can run the tests quickly on the command line with `yarn test Pharmacy` to make sure I haven't flubbed it again :)

Yeah, caught that.  I've updated them to (hopefully) be timezone agnostic so they'll pass for automated testing & any of us regardless of location

File: src/containers/PharmacyList/PharmacyList.test.tsx [hunk 2]
```diff
@@ -4,13 +4,20 @@ import { VaccineAvailabilityExpandedResponse } from "../../apiClient";
 import { server } from "../../mocks/server";
 import { render, screen } from "../../testUtils";
 import { PharmacyList } from "./PharmacyList";
+import { format } from "date-fns-tz";
+
+const formattedDate = format(
```
Comment (chrisahardie) [inline src/containers/PharmacyList/PharmacyList.test.tsx:9]: Awesome, now we'll get consistent results irrespective of the timezone of the machine the test is being run on

👍

**Review by chrisahardie — APPROVED** [review]

**Pull request merged**